### ISSUE TITLE
Support python uv as pip-compile compatible replacement

### ISSUE THREAD

**Issue by avilaton** [dependabot/dependabot-core#10039]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

We are trying to draft support for using https://github.com/astral-sh/uv as a replacement for pip-tools in dependabot.

The reason for this is that uv is much faster and many projects have already started switching to it. UV is a pip-tools compatible replacement written in rust.

If we are lucky, it should be as easy as changing `pip-compile --stuff` for `uv pip compile --stuff` and adding `uv` as one of the python helpers requirements.

**Comment by edgarrmondragon** [comment]
I think https://github.com/dependabot/dependabot-core/pull/10040 is ready for a review by the maintainers 🙂

**Comment by mistercrunch** [comment]
Linking my comment here https://github.com/astral-sh/uv/issues/5487#issuecomment-2407871061 around supporting multi-synchronized-outputs, which I think is super relevant to the uv/dependabot integration.

**Comment by EdmundGoodman** [comment]
In the related issue #10478, I have just but [the following comment](https://github.com/dependabot/dependabot-core/issues/10478#issuecomment-2434963744) on a work-around for getting fairly similar functionality to Dependabot in the meantime before these issue are resolved:

> This has also blocked us, so +1 for prioritising this.
> 
> As a stopgap in the meantime, I've hacked together a small GitHub Actions workflow which provides fairly similar functionality to unblock our project whilst we wait. A small demo is available here <https://github.com/EdmundGoodman/update-bot> if it is helpful to anyone else.
> 
> It slightly differs from dependabot in that it makes a PR on a cron schedule if any dependency can be updated rather than whenever a security vulnerability is found, but is good enough for us for now. It differs from [other workflows I've seen in this thread](https://github.com/cookiecutter/cookiecutter-django/pull/5422), as it PRs rather than just directly committing to main which could break things.

This might require a tiny bit of modification to work with `uv pip compile`, but the general idea is the same so thought I'd also mention it here in case it can help anyone.

**Comment by abdulapopoola** [comment]
Let's use #10478 to track this conversation. Closing this as a duplicate.

**Comment by alex** [comment]
It may look like a duplicate, but it's not.

`uv` has two "modes", one is its own one, which uses `uv.lock` and is tracked in #10478. This issue concerns using `uv pip compile`, which produces `requirements.txt` files which are identical in structure to what `pip-compile` produces.

**Comment by abdulapopoola** [comment]
thanks for the clarification @alex 

We're currently planning towards our upcoming quarter and one potential item for our roadmap is supporting `uv` as its full-fledged standalone ecosystem (i.e. providing support for all `uv` modes). Having a single issue to consolidate all discussions makes it easier for us to factor into plannning.

**Comment by alex** [comment]
Given that, I suggest titling that issue accordingly, right now it indicates a scope that does not encompass what this issue is about.

I don't work on dependabot, so y'all will know better than me, but based on my experience in the Python ecosystem, I think it's highly likely to be the case that there's almost no overlap in the engineering beteween supporting `uv` for updating `uv.lock` and supporting it for `uv pip compile`.

**Comment by ulgens** [comment]
Agreeing with @alex here, this is not a duplicate of #10478 and #10478 doesn't cover what is proposed here. 

Currently, dependabot is spending more than 10 minutes for a dependency list with 40~ entries. The majority of this time is used by pip-compile, regardless of the ecosystem. Having uv instead of pip-compile as the universal Python dependency manager would save 90% of runtime from all Python-related ecosystems.

**Comment by mistercrunch** [comment]
> Having uv instead of pip-compile as the universal Python dependency manager would save 90% of runtime from all Python-related ecosystems.

Amen!

**Comment by mistercrunch** [comment]
Also noting that the "magic" approach of trying to guess where files might be, and parsing comments left behind by `pip-compiile` or `uv pip compile` as to how these files were generated completely breaks down in more complex environments where multiple builds are required. In our case, we need different requirements sets for different envrionments, and these environments need to be kept in sync for the libs they share. A concrete example lives here https://github.com/apache/superset/blob/master/scripts/uv-pip-compile.sh , and it seems there's no way to tell dependabot to run this specific script "atomically". Also in our monorepo our various reqs file (.in and .txt) live in a subfolder as we have a monorepo. Would love for dependadbot to extend configurability to better support/configure around monorepos.

**Comment by ulgens** [comment]
@abdulapopoola Can you please reopen this issue?
